apps/jobbrowser/src/jobbrowser/templates/job_browser.mako

```diff
@@ -4080,18 +4082,27 @@ ${ commonheader("Job Browser", "jobbrowser", user, request) | n,unicode }
 
       var updateJobTimeout = -1;
       var updateJobsTimeout = -1;
+      var jobUpdateCounter = 0;
+      var exponentialFactor = 1;
       self.job.subscribe(function(val) {
         self.monitorJob(val);
       });
 
       self.monitorJob = function(job) {
         window.clearTimeout(updateJobTimeout);
         window.clearTimeout(updateJobsTimeout);
+        jobUpdateCounter = 0;
+        exponentialFactor = 1;
         if (self.interface() && self.interface() !== 'slas' && self.interface() !== 'oozie-info' && self.interface !== 'queries'){
           if (job) {
             if (job.apiStatus() === 'RUNNING') {
               var _updateJob = function () {
-                var def = job.updateJob();
+                jobUpdateCounter++;
+                var updateLogs = (jobUpdateCounter % exponentialFactor) === 0;
+                if(updateLogs && exponentialFactor < 50) {
```

Comment:
Please add a note about exponential backoff.

Reply:
Indeed a comment would be good, if I read this right it'll limit the updateLogs call to nth job update 1, 2, 4, 8, 16, 32, 64, 96...

Reply:
updateLogs will be true when jobUpdateCounter is at 1, 2, 4, 8, 16, 32, 64, 128, 192, 256....
Increasing exponentialFactor is caped above 50. So its value will be 1, 2, 4, 8, 16, 32, 64 and then remains at 64.
Its caped above 50 so that the delay between log updates doesn't become too big.

Should I add this to the commit message or the code?

Reply:
Ah yes, the last round it'll be 64. Now, every 64th is a bit high IMO, not sure what that translates to time.. Why not every 16th?

Reply:
JB_SINGLE_CHECK_INTERVAL_IN_MILLIS = 5sec. So 64 means a bit more than 5 minutes.
In one of the customer cluster log fetch was taking more than 2 minutes. So 64 sounded like a good number, as the next lower number 32 is just a bit more than 2.5 minutes.

Best would be if we can do a complete refactor and make log fetch independent of updateJob, and also start a log fetch only after one is complete. Right now if a log fetch takes longer other fetches could be fired in parallel.

Reply:
Yeah, that makes sense, only start the timer for the next fetch once complete.  But for now good I guess.